lib/Doctrine/ODM/MongoDB/UnitOfWork.php

```diff
@@ -1297,7 +1298,6 @@ private function cascadePostUpdateAndPostPersist(ClassMetadata $class, $document
                     } else {
                         if ( ! empty($entryClass->lifecycleCallbacks[Events::postUpdate])) {
                             $entryClass->invokeLifecycleCallbacks(Events::postUpdate, $entry);
-                            $this->recomputeSingleDocumentChangeSet($entryClass, $entry);
```

Comment:
But in that case, I'd think that a postPersist lifecycle callback could easily do the same and modify the document.